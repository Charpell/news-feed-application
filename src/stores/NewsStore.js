import { EventEmitter } from 'events';
import AppDispatcher from '../dispatcher/AppDispatcher';

class NewsStore extends EventEmitter {
	constructor(props) {
		super(props);
		this.newsSources = [];
		this.newsSource = [];
	}

	setSources(sources) {
		this.newsSources = sources;
		this.emit('change');
	}

	setSource(singleSource) {
		this.newsSource = singleSource;
		this.emit('change');
	}

	getSources() {
		return this.newsSources;
	}

	getSource() {
		return this.newsSource;
	}

	handleActions(action) {
		switch (action.type) {
			case "RECIEVE_SOURCES":
				this.setSources(action.sources);
				break;
			case "RECIEVE_SOURCE":
				this.setSource(action.singleSource);
				break;
		}
	}
}

const newsStore = new NewsStore();
AppDispatcher.register(newsStore.handleActions.bind(newsStore));
export default newsStore;
