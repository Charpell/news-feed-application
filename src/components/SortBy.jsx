import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import NewsStore from '../stores/NewsStore';

/**
 * Get sortBysAvailabe
 * @class SortBy
 * @extends {React.Component}
 */
class SortBy extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      sortBysAvailable: []
    };
  }

  componentWillMount() {
    NewsStore.getNewsSources().then((result) => {
      this.setState({
        sortBysAvailable: result.data.sources
        .filter(source => (source.id === this.props.sourceId)),
      });
    });
  }

  /**
   * Renders latest and top button respectively.
   * @returns {void}
   * @memberof SortBy
   */
  render() {
    let sortBysAvailable;
    let sortNews;
    if (this.state.sortBysAvailable[0]) {
      sortBysAvailable = this.state.sortBysAvailable[0].sortBysAvailable;
      sortNews = sortBysAvailable.map(sortBy => (
        <li key={sortBy}>
          <Link
            className="btn sortBys btn-sm"
            to={`/newsfeeds/${this.props
              .sourceId}/${sortBy}`}
          >{sortBy.toUpperCase()} NEWS</Link>
        </li>
      ));
    }
    return (
      <div>
        <ul className="sort pull-left">
          {sortNews}
        </ul>
      </div>);
  }


}

SortBy.propTypes = {
  sourceId: PropTypes.string
};

export default SortBy;
