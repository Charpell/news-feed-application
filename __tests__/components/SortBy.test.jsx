import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import SortBy from '../../src/components/SortBy';
import NewsStore from '../../src/stores/NewsStore';

describe('SortBy Page', () => {
  it('renders a snapshot', () => {
    const tree = renderer.create(<SortBy />).toJSON();
    expect(tree).toMatchSnapshot();
  });


  it('calls componentWillMount() lifecycle method', () => {
    const componentWillMountSpy = jest
    .spyOn(SortBy.prototype, 'componentWillMount');
    const wrapper = mount(<SortBy />);
    expect(wrapper).toBeDefined();
    expect(componentWillMountSpy).toHaveBeenCalled();
    expect(componentWillMountSpy).toHaveBeenCalledTimes(1);
  });
});
