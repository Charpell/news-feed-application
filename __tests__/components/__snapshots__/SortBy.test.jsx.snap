// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortBy Page renders a snapshot 1`] = `
<div>
  <ul
    className="sort pull-left"
  />
</div>
`;
